apps/frontend/src/pages/leaderboard/[chipIssuer].tsx

```diff
@@ -66,7 +68,138 @@ const LeaderboardPage: React.FC = () => {
+          </div>
+          <div className="justify-start items-start gap-2 flex">
+            <div className="text-right text-[#090909]/40 text-xs font-medium font-['DM Sans'] leading-none">Total taps
+            </div>
+          </div>
+        </div>
+      </div>
+
+      {
+        leaderboardEntries.entries.map((entry: LeaderboardEntry, index) => {
+          let position = index + 1;
+
+          // Handle ties
+          if (entry.tapCount == lastTapCount) {
+            // If equal, use tiedPosition (which is the first position, not last, of the tied entries)
+            position = tiedPosition;
+          } else {
+            // If not equal, update tiedPosition
+            tiedPosition = position;
+          }
+
+          // Update lastTapCount
+          lastTapCount = entry.tapCount
+
+
+          // Default styling
+          const styling: any = { // eslint-disable-line @typescript-eslint/no-explicit-any
+            "positionColor": "bg-black/20",
+            "positionTextColor": "",
+            "fontStyling": "text-[#090909]/60 font-sm",
+            "divider": false,
+          }
+
+          let username = entry.username;
+          const tapCount = entry.tapCount;
+
+          if (position == 1) {
+            styling.positionColor = "bg-[#090909]";
+            styling.positionTextColor = "text-white"; // Done
+            styling.fontStyling = "text-[#090909] font-medium";
+          }
+
+          if (entry.username == userLeaderboardPosition.username) {
+            styling.positionColor = "bg-[#f74227]";
+            styling.positionTextColor = "text-white"; // Done
+            styling.fontStyling = "text-[#090909] font-medium";
+            username += " (me)"; // Done
+
+            if (userLeaderboardPosition.leaderboardPosition != tiedPosition) {
+              // Update position if you're tied with another user
+              setLeaderboardPosition({username: entry.username, leaderboardPosition: tiedPosition})
+            }
+          }
+
+          if (index == 9) {
+            styling.divider = true;
+          }
+
+          return (
+            <div key={index}>
+            <div className="h-6 px-4 justify-between items-center inline-flex"
+                 style={{width: "100%", marginBottom: "4px", marginTop: "4px"} as CSSProperties}>
+              <div className="grow shrink basis-0 h-6 justify-start items-center gap-3 flex">
+                <div className={`w-10 h-6 px-0.5 py-2 ${styling.positionColor} rounded-[67px] justify-center items-center gap-2 flex`}>
+                  <div
+                    className={`text-center ${styling.positionTextColor} text-[#090909] text-sm font-medium font-['DM Sans'] leading-tight`}>{position}</div>
+                </div>
+                <div
+                  className={`grow shrink basis-0 ${styling.fontStyling} font-['DM Sans'] leading-tight`}>{username}
```

Comment:
I'd like to figure out how to apply `overflow: hidden` on this attribute but I need to define a max width somehow, I'm not sure how to set a max width that's appropriate for multiple phone screen types. This issue only applies to really long usernames, so it's not first priority.